cmake/HPX_AddModule.cmake

```diff
@@ -204,123 +201,112 @@ function(add_hpx_module name)
+  if(NOT "${modulename}" STREQUAL "config_registry")
+    target_link_libraries(hpx_${modulename} PUBLIC hpx_config_registry)
   endif()
 
   if(${_have_compatibility_headers_option}
-     AND HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS
+     AND HPX_${modulename_upper}_WITH_COMPATIBILITY_HEADERS
   )
     target_include_directories(
-      hpx_${name} PUBLIC $<BUILD_INTERFACE:${COMPAT_HEADER_ROOT}>
+      hpx_${modulename} PUBLIC $<BUILD_INTERFACE:${COMPAT_HEADER_ROOT}>
     )
   endif()
 
   target_compile_definitions(
-    hpx_${name} PRIVATE $<$<CONFIG:Debug>:DEBUG> $<$<CONFIG:Debug>:_DEBUG>
-                        HPX_EXPORTS
+    hpx_${modulename}
+    PRIVATE $<$<CONFIG:Debug>:DEBUG> $<$<CONFIG:Debug>:_DEBUG>
+            HPX_${libname_upper}_EXPORTS
   )
 
   # This is a temporary solution until all of HPX has been modularized as it
   # enables using header files from HPX for compiling this module.
-  target_include_directories(
-    hpx_${name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../..
-  )
+  target_include_directories(hpx_${modulename} PRIVATE ${HPX_SOURCE_DIR})
 
   add_hpx_source_group(
-    NAME hpx_{name}
+    NAME hpx_{modulename}
     ROOT ${HEADER_ROOT}/hpx
     CLASS "Header Files"
     TARGETS ${headers}
   )
   add_hpx_source_group(
-    NAME hpx_{name}
+    NAME hpx_{modulename}
     ROOT ${SOURCE_ROOT}
     CLASS "Source Files"
     TARGETS ${sources}
   )
   if(${_have_compatibility_headers_option}
-     AND HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS
+     AND HPX_${modulename_upper}_WITH_COMPATIBILITY_HEADERS
   )
     add_hpx_source_group(
-      NAME hpx_${name}
+      NAME hpx_${modulename}
       ROOT ${COMPAT_HEADER_ROOT}/hpx
       CLASS "Header Files"
       TARGETS ${compat_headers}
     )
   endif()
 
-  if(${name}_GLOBAL_HEADER_GEN OR ${name}_CONFIG_FILES)
+  if(${modulename}_GLOBAL_HEADER_GEN OR ${modulename}_CONFIG_FILES)
     add_hpx_source_group(
-      NAME hpx_{name}
+      NAME hpx_{modulename}
       ROOT ${CMAKE_CURRENT_BINARY_DIR}/include/hpx
       CLASS "Generated Files"
       TARGETS ${generated_headers}
     )
   endif()
-  if(${name}_FORCE_LINKING_GEN)
-    add_hpx_source_group(
-      NAME hpx_{name}
-      ROOT ${CMAKE_CURRENT_BINARY_DIR}/include/hpx
-      CLASS "Generated Files"
-      TARGETS ${force_linking_header}
-    )
-    add_hpx_source_group(
-      NAME hpx_{name}
-      ROOT ${CMAKE_CURRENT_BINARY_DIR}/src
-      CLASS "Generated Files"
-      TARGETS ${force_linking_source}
-    )
-  endif()
   add_hpx_source_group(
-    NAME hpx_{name}
+    NAME hpx_{modulename}
     ROOT ${CMAKE_CURRENT_BINARY_DIR}/include/hpx
     CLASS "Generated Files"
     TARGETS ${config_header}
   )
 
   set_target_properties(
-    hpx_${name} PROPERTIES FOLDER "Core/Modules" POSITION_INDEPENDENT_CODE ON
+    hpx_${modulename} PROPERTIES FOLDER "Core/Modules" POSITION_INDEPENDENT_CODE
+                                                       ON
```

Comment:
Please add a `FOLDER` property for the static libraries, possibly combining the shared and static libraries. For instance:
```
  set_target_properties(
    hpx_${modulename} PROPERTIES FOLDER "Core/Modules/${modulename}"
  )
  set_target_properties(
    hpx_${modulename}_static PROPERTIES FOLDER "Core/Modules/${modulename}"
  )
```

Reply:
Possibly even `"Core/Modules/Core/${modulename}"` and `"Core/Modules/Full/${modulename}"` as appropriate.